src/Microsoft.VisualStudio.Editor.Razor/BackgroundParser.cs

```diff
@@ -347,14 +349,31 @@ private RazorCodeDocument ParseChange(ITextSnapshot snapshot, CancellationToken
             {
                 EnsureOnThread();
 
-                var sourceDocument = new TextSnapshotSourceDocument(snapshot, _filePath);
-                var imports = _templateEngine.GetImports(_filePath);
+                var projectItem = new TextSnapshotProjectItem(snapshot, _projectDirectory, _relativeFilePath, _filePath);
```

Comment:
A little bit of work here... We need to make sure our `ITextSnapshot` => `RazorProjectItem` has sufficient information for any `RazorProjectEngine` configurations to act on.